packages/executor/src/executor.ts

```diff
@@ -549,6 +590,161 @@ export class BasicExecutor implements IExecutor {
+            }),
+            nonce,
+            gasPriceParameters.maxFeePerGas,
+            gasPriceParameters.maxPriorityFeePerGas,
+            "pending",
+            this.noEip1559Support,
+            this.customGasLimitForEstimation,
+            this.reputationManager,
+            childLogger
+        )
+
+        if (simulatedOps.length === 0) {
+            childLogger.warn("no ops to bundle")
+            this.markWalletProcessed(wallet)
+            return compressedOps.map((op) => {
+                return {
+                    success: false,
+                    error: {
+                        userOpHash: getUserOperationHash(op.inflatedOp, entryPoint, this.walletClient.chain.id),
+                        reason: "INTERNAL FAILURE"
+                    }
+                }
+            })
+        }
+
+        // if not all succeeded, return error
+        if (simulatedOps.some((op) => op.reason !== undefined)) {
+            childLogger.warn("some ops failed simulation")
+            this.markWalletProcessed(wallet)
+            return simulatedOps.map((op) => {
+                return {
+                    success: false,
+                    error: {
+                        userOpHash: op.op.userOperationHash,
+                        reason: op.reason as string
+                    }
+                }
+            })
+        }
+
+        opsToBundle = simulatedOps.filter((op) => op.reason === undefined).map((op) => op.op)
+
+        let txHash: HexData32
+        try {
+            const gasOptions = this.noEip1559Support
+                ? {
+                    gasPrice: gasPriceParameters.maxFeePerGas,
+                }
+                : {
+                    maxFeePerGas: gasPriceParameters.maxFeePerGas,
+                    maxPriorityFeePerGas: gasPriceParameters.maxPriorityFeePerGas,
+                };
+
+            // need to use sendTransction to target BundleBulker's fallback
+            txHash = await this.walletClient.sendTransaction({
+                account: wallet,
+                to: this.compressionHandler.bundleBulkerAddress,
+                data: createCompressedCalldata(compressedOps, this.compressionHandler.perOpInflatorId),
+                gas: gasLimit,
+                nonce: nonce,
+                ...gasOptions
+            })
+        } catch (err: unknown) {
+            sentry.captureException(err)
+            childLogger.error({ error: JSON.stringify(err) }, "error submitting bundle transaction")
+            this.markWalletProcessed(wallet)
+            return []
+        }
+
+        const userOperationInfos = opsToBundle.map((op) => {
+            this.metrics.userOperationsSubmitted.inc()
+
+            return {
+                mempoolUserOperation: op.userOperation,
+                userOperationHash: op.userOperationHash,
+                lastReplaced: Date.now(),
+                firstSubmitted: Date.now()
+            }
+        })
+
+        const bb = getContract({
```

Comment:
do we really need to create a contract instance here if we just need its address?